File: diploma.client/src/models/ConsensusEvaluation.ts
import Product from "./Product";

export default class ConsensusEvaluation {
    consensusEvaluationId: number;
    priceStrategy: number;
    demand: number;
    quality: number;
    priceQuality: number;
    timestamp: Date;
    productId: string;
    product?: Product | null;

    constructor() {
        this.consensusEvaluationId = 0;
        this.productId = "";
        this.product = null;
        this.priceStrategy = 0;
        this.demand = 0;
        this.quality = 0;
        this.priceQuality = 0;
        this.timestamp = new Date();
    }
}

File: Diploma.Server/Models/ConsensusEvaluation.cs
namespace Diploma.Server.Models
{
    public class ConsensusEvaluation
    {
        public int ConsensusEvaluationId { get; set; }
        public double PriceStrategy { get; set; }
        public double Demand { get; set; }
        public double Quality { get; set; }
        public double PriceQuality { get; set; }
        public DateTime? Timestamp { get; set; }
        public required string ProductId { get; set; }
        public required Product Product { get; set; }
    }
}
